#! /usr/bin/env python

PACKAGE='robot_movement_control'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("mEnabled", bool_t, 0, "Enabled", True)
		
gen.add("GoalZ", double_t, 0, "Goal Z value", 1.0, -5.0, 5.0)
gen.add("GoalX", double_t, 0, "Goal X value", 0.0, -5.0, 5.0)
gen.add("ZScale", double_t, 0, "Z Scale Factor", 0.95, 0.0, 5.0)
gen.add("XScale", double_t, 0, "X Scale Factor", 0.9, 0.0, 5.0)
gen.add("GoalInterval", double_t, 0, "Goal min interval", 0.01, 0.0, 5.0)
gen.add("DamperDivisor", double_t, 0, "Weighted Damper", 1.0, 0.0, 30.0)

gen.add("MaxRobotSpeed", double_t, 0, "Max linear speed", 1.0, 0.0, 10.0)

exit(gen.generate(PACKAGE, "robot_movement_control", "movement_control"))

